compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -1797,11 +1821,21 @@ object TypeComparer {
 
   val NoApprox: ApproxState = new ApproxState(0)
 
+  def explain[T](say: String => Unit)(op: Context => T)(implicit ctx: Context): T = {
+    val (res, explanation) = underlyingExplained(op)
```

Comment:
I have the impression this is could be simpler. Why not work with the existing `explained`? It seems the added complications are not needed.

Reply:
See https://github.com/lampepfl/dotty/pull/5611/commits/d1fa5d73c4194448e1ab0d4d4e39a764b6e1f651#r242125874.